### PR TITLE
docs: document_loaders classification

### PR DESCRIPTION
**Pull request by leo-gan**
**Problem statement:** the [document_loaders](https://python.langchain.com/en/latest/modules/indexes/document_loaders.html#) section is too long and hard to comprehend.
**Proposal:** group document_loaders by 3 classes: (see `Files changed` tab)

UPDATE: I've completely reworked the document_loader classification.
Now this PR changes only one file! 

FYI @eyurtsev @hwchase17

### CODE REVIEW

File: docs/modules/indexes/retrievers/examples/chatgpt-plugin-retriever.ipynb [hunk 1]
```diff
@@ -37,7 +37,7 @@
     "# This is from https://langchain.readthedocs.io/en/latest/modules/document_loaders/examples/csv.html\n",
     "\n",
     "from langchain.document_loaders.csv_loader import CSVLoader\n",
-    "loader = CSVLoader(file_path='../../document_loaders/examples/example_data/mlb_teams_2012.csv')\n",
+    "loader = CSVLoader(file_path='../../document_loaders/examples/../example_data/mlb_teams_2012.csv')\n",
```
Comment (hwchase17) [inline docs/modules/indexes/retrievers/examples/chatgpt-plugin-retriever.ipynb:40]: weird pathing

File: docs/modules/indexes/document_loaders/examples/example_data/notebook.ipynb [hunk 2]
```diff
@@ -25,7 +25,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "loader = NotebookLoader(\"example_data/notebook.ipynb\")"
+    "loader = NotebookLoader(\"../../../example_data/notebook.ipynb\")"
```
Comment (hwchase17) [inline docs/modules/indexes/document_loaders/examples/example_data/notebook.ipynb:28]: this shouldnt really need to change?

**Review by hwchase17 — COMMENTED** [review]
hmmm in theory i like this, but the distinction seems a bit blurred. for example, why is Google Drive a formatter and not a knowledge document loader

File: docs/modules/indexes/retrievers/examples/chatgpt-plugin-retriever.ipynb (hunk 1 above)
Comment (leo-gan) [reply docs/modules/indexes/retrievers/examples/chatgpt-plugin-retriever.ipynb:40]: OOps. My bad.

File: docs/modules/indexes/document_loaders/examples/example_data/notebook.ipynb (hunk 2 above)
Comment (leo-gan) [reply docs/modules/indexes/document_loaders/examples/example_data/notebook.ipynb:28]: No. We don't need to change test data. Thanks.

> hmmm in theory i like this, but the distinction seems a bit blurred. for example, why is Google Drive a formatter and not a knowledge document loader

I'll add more description to this. 
The idea is "knowledge loader" works with storage that we do not control. Something that can be used as a "tool" (in terms of LangChain). That can be accessed with queries. Something that can be considered as a source of "external" knowledge. We can allow LLM to make queries and get information or we can download documents and use them in a more controllable way.
"Formatters" can be as easy as transformers for CSV, SQL, etc. But they also can be cloud services or app stores. They can be hosted out of our control but the information inside is under our control.

**Comment by hwchase17** [comment]
> The idea is "knowledge loader" works with storage that we do not control. Something that can be used as a "tool" (in terms of LangChain). That can be accessed with queries. Something that can be considered as a source of "external" knowledge. We can allow LLM to make queries and get information or we can download documents and use them in a more controllable way.
"Formatters" can be as easy as transformers for CSV, SQL, etc. But they also can be cloud services or app stores. They can be hosted out of our control but the information inside is under our control

hmm i think formatter and i think csv or word... but not like google drive. like google drive could have csv files in it

i would be down to split out the ones which related to a certain file type. eg csv/pdf/ppt/etc. and then other ones could load in from various locations (eg from drive or website etc) and use formatters under the hood

this may be related to some of the stuff @eyurtsev is working on?

**Comment by hwchase17** [comment]
> > 
> 
> How about splitting it into 3 classes?
> 
> * `formatters`: CSV, PDF, ...
> * `controllable sources`: Google Drive, Microsoft Word, Facebook Chat, ...
> * `external sources`:  Guttenberg, iFixit, ...
>   I still don't like the class names. That means the `mental picture` is not good

what is definition of those categories? eg why is microsoft word (.docx) not a format?

**Comment by leo-gan** [comment]
> > > 
> > 
> > 
> > How about splitting it into 3 classes?
> > 
> > * `formatters`: CSV, PDF, ...
> > * `controllable sources`: Google Drive, Microsoft Word, Facebook Chat, ...
> > * `external sources`:  Guttenberg, iFixit, ...
> >   I still don't like the class names. That means the `mental picture` is not good
> 
> what is definition of those categories? eg why is microsoft word (.docx) not a format?

@hwchase17  I've completely reworked the document_loader classification. Please, check it out.
One good side effect: Now this PR changes only one file.

**Comment by leo-gan** [comment]
@hwchase17 any comments? If you are busy, maybe @dev2049 can help? TNX

**Review by hwchase17 — APPROVED** [review]
awesome - thanks!

**Pull request merged**